data/transactions/logic/eval.go

```diff
@@ -371,8 +426,7 @@ func feeCredit(txgroup []transactions.SignedTxnWithAD, minFee uint64) uint64 {
 	}
 	// Overflow is impossible, because TxnGroupBatchVerify checked.
 	feeNeeded := minFee * minFeeCount
-
-	return feesPaid - feeNeeded
+	return basics.SubSaturate(feesPaid, feeNeeded)
```

Comment:
At this point, we know `feesNeeded` **cannot** be greater than `feedsPaid`, correct?

Reply:
@jasonpaulos I think you raise a good question.  I don't feel confident that I can address the question off-hand.  But, I see a related, minor improvement.  Can you review https://github.com/algorand/go-algorand/pull/4647?

Reply:
In the real running algod, yes, we can be sure, as per the comment `// Overflow is impossible, because TxnGroupBatchVerify checked.`

But, in some tests, that function is not called, so I ended up losing some time to an overflow here while debugging a different problem. I figured I would prevent future me from having the same problem.